test/projectManager.ts

```diff
@@ -113,28 +135,79 @@ export class ProjectManager {
     /**
      * Adds a plugin to a Cordova project.
      */
-    public static addPlugin(projectFolder: string, plugin: string): Q.Promise<void> {
+    public static addPlugin(projectFolder: string, plugin: string): Q.Promise<string> {
         return ProjectManager.execAndLogChildProcess("cordova plugin add " + plugin, { cwd: projectFolder });
     }    
 
     /**
      * Builds a specific platform of a Cordova project. 
      */
-    public static buildPlatform(projectFolder: string, targetPlatform: platform.IPlatform): Q.Promise<void> {
-        return ProjectManager.execAndLogChildProcess("cordova build " + targetPlatform.getCordovaName(), { cwd: projectFolder });
+    public static buildPlatform(projectFolder: string): Q.Promise<string> {
+        var targetPlatform = platform.PlatformResolver.resolvePlatform(tu.TestUtil.readTargetPlatform());
+        
+        // don't print output here because the iOS build process outputs so much nonsense that it buffer overflows and exits the entire test process
+        return ProjectManager.execAndLogChildProcess("cordova build " + targetPlatform.getCordovaName(), { cwd: projectFolder }, false);
     }
     
     /**
      * Prepares a specific platform of a Cordova project. 
      */
-    public static preparePlatform(projectFolder: string, targetPlatform: platform.IPlatform): Q.Promise<void> {
+    public static preparePlatform(projectFolder: string): Q.Promise<string> {
+        var targetPlatform = platform.PlatformResolver.resolvePlatform(tu.TestUtil.readTargetPlatform());
+        
         return ProjectManager.execAndLogChildProcess("cordova prepare " + targetPlatform.getCordovaName(), { cwd: projectFolder });
     }
 
     /**
-     * Runs a project to a given target / platform.
+     * Launch the test app on the given target / platform.
+     */
+    public static launchApplication(namespace: string): Q.Promise<string> {
+        var targetPlatform = platform.PlatformResolver.resolvePlatform(tu.TestUtil.readTargetPlatform());
+        
+        var emulatorManager = targetPlatform.getEmulatorManager();
+        if (emulatorManager) {
```

Comment:
I would remove the check for `emulatorManager` in all these methods, now that we have an emulator manager for both platforms.